packages/discord.js-utilities/src/lib/PaginatedMessage.ts

```diff
@@ -1,65 +1,162 @@
 import type { APIMessage, User, TextChannel, NewsChannel, Message, MessageReaction, ReactionCollector } from 'discord.js';
 
+/**
+ * This is a [[PaginatedMessage]], a utility to paginate messages (usually embeds).
+ * You must either use this class directly or extend it.
+ *
+ * [[PaginatedMessage]] uses actions, these are essentially reaction emojis, when triggered run the said action.
+ * You can utilize your own actions, or you can use the [[PaginatedMessage.defaultActions]].
+ * [[PaginatedMessage.defaultActions]] is also static so you can modify these directly.
+ *
+ * [[PaginatedMessage]] also uses pages, these are simply {@link https://discord.js.org/#/docs/main/stable/class/APIMessage APIMessages}.
+ *
+ * @example
+ * ```typescript
+ * const handler = new PaginatedMessage();
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // To utilize actions you can use the IPaginatedMessageAction by implementing it into a class.
+ *
+ * class ForwardAction implements IPaginatedMessageAction {
```

Comment:
This presents a class...however, said class would need to be instantiated *somewhere and by someone*

Reply:
Yes, that's correct which is why if you read further in that same code block it states it can also pass an object.

Reply:
Doesn't change the fact you're presenting the possibility of people using raw classes in addAction, which is not handled. Either document that you NEED to pass an instance yourself, or handle it in the code

Reply:
What do you think `IPaginatedMessageAction` is? It's an interface, meaning you can implement it into a class or directly an object. It clearly states this in the documentation. Your reasoning does not make sense imo.

Reply:
`class X implements IPaginatedMessageAction {}` only means that instances of `X` are assignable to the type `IPaginatedMessageAction`, not `X` itself.